src/components/ToDoList.vue

```diff
@@ -1,45 +1,68 @@
 <template>
-	<div>
-		<!-- <Header @toggle-visibility="toggleVisibility"></Header> -->
-		<div :style="{ display: shouldShow ? 'block' : 'none' }">
-			<div id="clear">
-				<button @click="clear">Clear</button>
-			</div>
-			<ToDoForm @add-todo="addTodo" />
-		</div>
-		<ToDoItem :todos="todos" @delete-item="removeItem" />
-	</div>
+  <div
+    class="w-1/2 h-screen mx-auto max-w-xl min-w-[375px] phone:bg-red-200 desktop:bg-blue-200"
+  >
+    <div class="px-5 h-screen">
+      <Header @toggle-visibility="toggleVisibility"></Header>
+      <div class="flex justify-center align-center">
+        <div
+          :class="isNewElementFormActive ? 'block' : 'hidden'"
+          class="mb-10 border border-black rounded-xl border-2"
+        >
+          <ToDoForm @add-todo="addTodo" />
+        </div>
+      </div>
+      <ToDoItem :todos="todos" @delete-item="removeItem" />
+      <div class="flex justify-center align-middle"></div>
+      <div
+        :class="shouldEmptyListImageBeDisplayed ? 'block' : 'hidden'"
+        class="flex justify-center align-center"
+      >
+        <EmptyListImage />
+      </div>
+      <div class="flex justify-center align-center">
+        <button
+          class="bg-transparent p-2 rounded-md mb-2 border border-black"
+          @click="clear"
+        >
+          Clear
+        </button>
+      </div>
+    </div>
+  </div>
 </template>
 
 <script setup lang="ts">
-	import { ref } from 'vue'
-	import ToDoForm from './ToDoForm.vue'
-	//import Header from './Header.vue'
-	import ToDoItem from './ToDoItem.vue'
+import { computed, reactive, ref } from 'vue'
+import ToDoForm from './ToDoForm.vue'
+import Header from './Header.vue'
+import ToDoItem from './ToDoItems.vue'
+import { Todo } from './types'
+import EmptyListImage from './EmptyListImage.vue'
 
-	interface Todo {
-		title: string;
-		priority: string;
-		text: string;
-	}
+const shouldEmptyListImageBeDisplayed = computed(
+  () => !isNewElementFormActive.value && !todos.length
+)
 
-	const shouldShow = ref(true);
+const isNewElementFormActive = ref(true)
 
-	const todos = ref<Todo[]>([]);
+const todos = reactive<Todo[]>([])
 
-	function addTodo(todo: Todo) {
-		todos.value.push(todo);
-	}
+function addTodo(todo: Todo) {
+  const todoCopy = { ...todo }
```

Comment:
If you want to clone an object always use the built in structuredClone function, https://developer.mozilla.org/en-US/docs/Web/API/structuredClone